c/tests/test_tables.c

```diff
@@ -4843,25 +4814,23 @@ test_ibd_finder_multiple_trees(void)
     tsk_treeseq_t ts;
     tsk_table_collection_t tables;
     tsk_id_t samples[] = { 0, 1, 0, 2 };
-    tsk_ibd_finder_t ibd_finder;
+    tsk_id_t *pair;
+    tsk_ibd_result_t result;
     double true_left[2][2] = { { 0.0, 0.7 }, { 0.7, 0.0 } };
     double true_right[2][2] = { { 0.7, 1.0 }, { 1.0, 0.7 } };
     double true_node[2][2] = { { 4, 5 }, { 5, 6 } };
     tsk_segment_t *seg = NULL;
 
-    // Read in the tree sequence.
     tsk_treeseq_from_text(&ts, 2, multiple_tree_ex_nodes, multiple_tree_ex_edges, NULL,
         NULL, NULL, NULL, NULL, 0);
     ret = tsk_treeseq_copy_tables(&ts, &tables, 0);
     CU_ASSERT_EQUAL_FATAL(ret, 0);
 
-    // Run ibd_finder.
-    ret = ibd_finder_init_and_run(&ibd_finder, &tables, samples, 2, 0.0, DBL_MAX);
+    ret = tsk_table_collection_find_ibd(&tables, &result, samples, 2, 0.0, DBL_MAX, 0);
     CU_ASSERT_EQUAL_FATAL(ret, 0);
-
-    // Check the output.
-    for (j = 0; j < (int) ibd_finder.num_pairs; j++) {
-        ret = tsk_ibd_finder_get_ibd_segments(&ibd_finder, j, &seg);
+    for (j = 0; j < (int) result.num_pairs; j++) {
+        pair = samples + 2 * j;
```

Comment:
I'm a bit confused about what this pair is supposed to be?

Reply:
It's accessing the jth pair, which is at offset 2 * j since each pair takes up two slots in the array. This is the same thing as saying ``pair = &samples[2 * j]``, as arrays are just pointer arithmetic in C.